core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -228,23 +226,26 @@ class KafkaServer(
         info(s"Cluster ID = $clusterId")
 
         /* load metadata */
-        val (preloadedBrokerMetadataCheckpoint, initialOfflineDirs) =
-          BrokerMetadataCheckpoint.getBrokerMetadataAndOfflineDirs(config.logDirs, ignoreMissing = true, kraftMode = false)
-
-        if (preloadedBrokerMetadataCheckpoint.version != 0) {
-          throw new RuntimeException(s"Found unexpected version in loaded `meta.properties`: " +
-            s"$preloadedBrokerMetadataCheckpoint. Zk-based brokers only support version 0 " +
-            "(which is implicit when the `version` field is missing).")
+        val initialMetaPropsEnsemble = {
+          val loader = new MetaPropertiesEnsemble.Loader()
+          config.logDirs.foreach(loader.addLogDir(_))
+          loader.load()
         }
```

Comment:
Shouldn't we be generating missing directory IDs here? As we do in KafkaRaftServer, since LogManager no longer does that.